ls1intum/Artemis
# `Quiz exercises`: Fix stuck loading cycle for competency selector in quiz exercise edit view

### Checklist
#### General


#### Client

### Motivation and Context



### Description


### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course *WITHOUT* Competencies

1. Log in to Artemis
2. Navigate to Course Administration
For each Exercise type:
3. Create Exercise
4. Check if you see a loading animation for the competency selector. Either the selector disappears after a short time OR it disappeared so quickly you didn't have time to see it. Both cases are fine.
5. Continue with the next Exercise type (step 3.) until all are covered



### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [competency-selection.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5747/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/competency-selection/competency-selection.component.ts.html) | 88.63% | ✅ |


### Screenshots
